src/app/ui-library/sheet.tsx

```diff
@@ -0,0 +1,99 @@
+import { XCircle } from '@phosphor-icons/react';
+import { isInOptions } from '@utils/helper';
+import Modal from 'react-modal';
+import styled, { useTheme } from 'styled-components';
+
+const BottomModalTitleText = styled.h1((props) => ({
+  ...props.theme.typography.body_bold_l,
+  flex: 1,
+}));
+
+const RowContainer = styled.div((props) => ({
+  display: 'flex',
+  flexDirection: 'row',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  margin: props.theme.space.m,
+}));
+
+const ButtonImage = styled.button({
+  backgroundColor: 'transparent',
+});
+
+const CustomisedModal = styled(Modal)`
+  overflow-y: auto;
+  position: absolute;
+`;
+
+const BodyContainer = styled.div`
+  margin: ${(props) => props.theme.space.m};
+`;
+
+type Props = {
+  title: string;
+  visible: boolean;
+  children: React.ReactNode;
+  onClose: () => void;
+  overlayStylesOverriding?: {};
+  contentStylesOverriding?: {};
+  className?: string;
+};
+
+function Sheet({
+  title,
+  children,
+  visible,
+  onClose,
+  overlayStylesOverriding,
+  contentStylesOverriding,
+  className,
+}: Props) {
+  const theme = useTheme();
+  const isGalleryOpen = isInOptions();
+
+  const customStyles = {
+    overlay: {
+      backgroundColor: theme.colors.background.modalBackdrop,
+      zIndex: 15000,
```

Comment:
lol. this is from the old sheet component right?

Reply:
Yeah, I think so 😅

Reply:
Do you think I should lower it? Not sure why it was so high in the first place, maybe there was something with a 10k z-index 😅

Reply:
The alternative is to use a proper modal implementation where we inject the modal into a parent that's on the base body tag. That should ignore the z-indexes of the other things in the main body container.